src/main/java/com/zufar/icedlatte/auth/google/endpoint/AuthEndpoint.java

```diff
@@ -0,0 +1,189 @@
+
+  @Autowired Environment environment;
+
+  @Value("${google.auth.server.url}")
+  public String authorizationServerUrl;
+
+  @Value("${google.client-id}")
+  public String clientId;
+
+  @Value("${google.client-secret}")
+  public String clientSecret;
+
+  @Value("${google.scope}")
+  public String scope;
+
+  @Value("${google.authenticateUri}")
+  public String authenticateUri;
+
+  @Value("${google.redirectUri}")
+  String redirectUri;
+
+  public static final String GOOGLE_AUTH = "/api/v1/3part-auth/google";
+
+  private final UserRepository userCrudRepository;
+  private final UserRegistrationService userRegistrationService;
+
+  @GetMapping()
+  public ResponseEntity<?> googleAuth() {
+
+    log.info("Received the request to add a google auth");
+    String redirectAuthUri =
+        "http://localhost"
+            + ":"
+            + environment.getProperty("server.port")
+            + GOOGLE_AUTH
+            + "/callback";
+    String url =
+        authorizationServerUrl
+            + "?"
+            + "scope="
+            + scope
+            + "&"
+            + "access_type=offline&"
+            + "include_granted_scopes=true&"
+            + "response_type=code&"
+            + "state=state_parameter_passthrough_value&"
+            + "redirect_uri="
+            + redirectAuthUri
+            + "&"
+            + "client_id="
+            + clientId;
+    return ResponseEntity.status(HttpStatus.TEMPORARY_REDIRECT).header("Location", url).build();
+  }
+
+  @GetMapping("/callback")
+  public ResponseEntity<?> googleAuthCallback(@RequestParam("code") String code)
+      throws IOException, GeneralSecurityException, InterruptedException {
+    log.warn("Received callback the request a google auth");
+
+    // get token
+    AuthorizationCodeFlow flow = initializeFlow();
+    String redirectAuthUri =
+        "http://localhost"
+            + ":"
+            + environment.getProperty("server.port")
+            + GOOGLE_AUTH
+            + "/callback";
+    TokenResponse token = flow.newTokenRequest(code).setRedirectUri(redirectAuthUri).execute();
+
+    HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    JsonFactory jsonFactory = GsonFactory.getDefaultInstance();
+    GoogleIdTokenVerifier verifier =
+        new GoogleIdTokenVerifier.Builder(httpTransport, jsonFactory)
+            .setAudience(Collections.singletonList(clientId))
+            .build();
+
+    // verify idToken
+    GoogleIdToken idToken = verifier.verify((String) token.get("id_token"));
+    if (idToken != null) {
+      GoogleIdToken.Payload payload = idToken.getPayload();
+      String email = payload.getEmail();
+      log.info("User email: {}", email);
+
+      // find or create user
+      String password = findOrCreateUser(email);
+      log.info("Success logged with email={}", email);
+
+      // login in main form
+      sendPostRequest(authenticateUri, new UserAuthenticationRequest(email, password));
```

Comment:
Is it necessary to send request? 🤔 
Is it possible to just leverage the `UserAuthenticationService`?

Reply:
no, its send post as frondend